src/expression.c

```diff
@@ -2761,6 +2761,29 @@ Expression *Expression::toBoolean(Scope *sc)
             goto Lagain;
         }
     }
+    // Other non-scalar types that are valid for bool conditions
+    // should have their contents lowered to valid scalars.
+    else if (tb->ty == Tarray)
```

Comment:
Pity. It doesn't make sense to put the comment anywhere else.

I'll just remove the ``else`` then...